modules/shortcodes/email-subscribe.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Class Jetpack_Email_Subscribe
+ * This class encapsulates:
+ * - a shortcode for subscribing to a MailChimp list.
+ * - a Gutenberg block for subscribing to a MailChimp list
+ * Both the shortcode and a block display a simple signup form with an "email" field.
+ * Submitting it subscribes the user to a MailChimp list selected in the "Sharing" section of Calypso.
+ * Other Email services and blocks can be implemented as well in the future.
+ */
+class Jetpack_Email_Subscribe {
+
+	private static $shortcode = 'jetpack-email-subscribe';
+
+	private static $css_classname_prefix = 'jetpack-email-subscribe';
+
+	private static $option_name = 'jetpack_mailchimp';
+
+	private static $block_name = 'mailchimp';
+
+	private static $instance;
+
+	private static $version = '1.0';
+
+	private function __construct() {
+	}
+
+	/**
+	 * This follows a classic singleton pattern.
+	 *
+	 * @return Jetpack_Email_Subscribe|null
+	 */
+	public static function get_instance() {
+		// Do not load this at all if it's neither a WPCOM or a connected JP site.
+		if ( ! ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) || Jetpack::is_active() ) ) {
+			return null;
+		}
+
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+
+		return self::$instance;
+	}
+
+	private function register_scripts_and_styles() {
+		wp_register_script( 'jetpack-email-subscribe', Jetpack::get_file_url_for_environment( '_inc/build/shortcodes/js/jetpack-email-subscribe.min.js', 'modules/shortcodes/js/jetpack-email-subscribe.js' ), array( 'jquery' ), self::$version );
+		wp_register_style( 'jetpack-email-subscribe', plugins_url( '/css/jetpack-email-subscribe.css', __FILE__ ), array(), self::$version );
+	}
+
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+		add_action( 'jetpack_options_whitelist', array( $this, 'filter_whitelisted_options' ), 10, 1 );
+		add_action( 'jetpack_blocks_to_register', array( $this, 'prevent_jetpack_register_block' ), 10, 1 );
+	}
+
+	/**
+	 * There is a structural problem in Jetpack_Gutenberg class that breaks server-side rendered blocks on WPCOM.
+	 * The problem stems from register_blocks being called very late. Too late in fact to WP_REST_Block_Renderer_Controller be aware of them.
+	 * Regular 'register_block_type' call does not register blocks as available from Calypso SDK.
+	 * I tried registering block twice, both with jetpack_register_block and register_block_type, but that produces a notice.
+	 * I introduced this callback and 'jetpack_blocks_to_register' filter as a way of dealing with this issue
+	 * @param $blocks - list of blocks that will be passed to 'register_block_type' call of Jetpack_Gutenberg.
+	 *
+	 * @return array
+	 */
+	public function prevent_jetpack_register_block( $blocks ) {
```

Comment:
We are doing something wrong here. We should always register the block using jetpack_register_block 
If you have specific visibility requirement that depend blog info (options, plan etc) You will need to use the callback. See https://github.com/Automattic/jetpack/blob/master/modules/simple-payments/simple-payments.php#L61 as an example.

Reply:
I believe this was discussed here:
https://github.com/Automattic/jetpack/pull/10597#discussion_r245018740